lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -448,3 +448,17 @@ <h4 class="message-title">${_('Your current verification will expire soon.')}</h
 <%static:require_module_async module_name="js/dateutil_factory" class_name="DateUtilFactory">
     DateUtilFactory.transform(iterationKey=".localized-datetime");
 </%static:require_module_async>
+
+
+<%static:require_module_async module_name="js/commerce/track_ecommerce_events" class_name="TrackECommerceEvents">
+  var link = $("#ecommerce_upgrade_link");
+  var event = function() {
+    TrackECommerceEvents.trackUpsellClick('course_dashboard_green', {
+      pageName: "course_dashboard",
+      linkType: "button",
+      linkCategory: "green_upgrade"
+    });
+  }
+
+  link.off('click').on('click', event);
```

Comment:
This page appears to be loading the async require modules twice and we want the event to be attached to the link once / fire once per click. This is the current workaround, but unfortunately the `off()` method would remove any existing events attached to the link before adding this one. I'm not familiar enough with all the locations that can potentially add events to the link, but this would affect anyone else attempting something similar, so  any suggestions on how to improve this would be appreciated!

Reply:
`link.off('click', '#ecommerce_upgrade_link', event)` might work?

Reply:
Tl;dr - we need to use `bind` and `unbind` to attach events to html elements